server/src/main/scala/org/ensime/model/ModelBuilders.scala

```diff
@@ -39,29 +39,33 @@ trait ModelBuilders { self: RichPresentationCompiler =>
     }
   }
 
-  def locateSymbolPos(sym: Symbol, needPos: PosNeeded): Option[SourcePosition] =
-    if (sym == NoSymbol || needPos == PosNeededNo) None
+  def locateSymbolPos(sym: Symbol, needPos: PosNeeded): Option[SourcePosition] = {
+    if (sym == NoSymbol || needPos == PosNeededNo)
+      None
     else if (sym.pos != NoPosition) {
-      if (needPos eq PosNeededYes)
+      if (needPos == PosNeededYes || needPos == PosNeededAvail) {
```

Comment:
Hmm, good point - I had considered the calculation performance cost, but not the message sizes/performance impact there. I will have a play.